compiler/hieFile/HieTypes.hs

```diff
@@ -127,7 +127,7 @@ data HieType a
   | HAppTy a (HieArgs a)
   | HTyConApp IfaceTyCon (HieArgs a)
   | HForAllTy ((Name, a),ArgFlag) a
-  | HFunTy  a a
+  | HFunTy a a a
```

Comment:
So, same as the first question: is `HFunTy` supposed to resemble `FunTy` or `HsFunTy`? In the latter case, we'd have to put something like an `HsArrow` as this extra argument.